lib/rubocop/cop/netlify/require_scope_base.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Netlify
+      class RequireScopeBase < Cop
+        RESTRICT_ON_SEND = [:require_scope, :public, :private, :protected]
+        
+        def on_class(node)
+          @require_scopes = []
+          @method_protection = :public
+          @is_controller = node.identifier.short_name.to_s.end_with?("Controller")
+        end
+        
+        def on_send(node)
+          if node.method_name == :require_scope
+            scopes = []
+            limits = {}
+            node.arguments.each do |option|
+              if option.is_a? RuboCop::AST::StrNode
+                scopes << option.value
+              elsif option.is_a? RuboCop::AST::HashNode
+                option.pairs.each do |pair|
+                  if pair.value.is_a? RuboCop::AST::ArrayNode
+                    limits[pair.key.value] = pair.value.values.map(&:value)
+                  elsif pair.value.is_a? RuboCop::AST::SymbolNode
+                    limits[pair.key.value] = [pair.value.value]
+                  end
+                end
+              end
+            end
+
+            @require_scopes << {
+              scopes: scopes,
+              limits: limits
+            }
+          else
+            @method_protection = node.method_name
+          end
+        end
+
+        private
+
+        def all_method_names_for_scope_checking
+          [:fake_method_for_scope_checking] + @require_scopes.map { |rs| (rs[:limits][:only] || []) +(rs[:limits][:except] || [])  }.reduce([], &:+).uniq
```

Comment:
What is `fake_method_for_scope_checking` needed for??

Reply:
Yeah. At least it needs a better name, and code comments in the source. It's what I came up for for testing `except:` clauses without making the code too complex. I'd wanted to add offensives at the moment of encountering the require_scope node, which is ahead of knowing what the method definitions are. I could drop this if rubocop supported two pass behaviors, or if there's an on_class_end equivalent _and_ it's legal to add offenses from prior nodes. 🤔

Reply:
Sorry, not sure I follow.
Can you reword the problem you're to solve? Maybe share and exanple?

Reply:
Resolved. Previously the offense could be added before seeing any method definitions. Now the offense will only be added once a method defintion with multiple matches is encountered. The code is easier to read and the badly named `fake_method_for_scope_checking` is no longer needed.